### PR TITLE
[SPARK-43064][SQL] Spark SQL CLI SQL tab should only show once statement once

### PR DESCRIPTION
**Pull request by AngersZhuuuu**
### What changes were proposed in this pull request?
Before 
截屏2023-04-07 下午4 22 54

After 
截屏2023-04-07 下午4 24 02

### Why are the changes needed?
Don't need show twice, too weird

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
MT

### CODE REVIEW

**Comment by AngersZhuuuu** [comment]
ping @cloud-fan

File: sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/SparkSQLDriver.scala [hunk 1]
```diff
@@ -65,8 +66,13 @@ private[hive] class SparkSQLDriver(val context: SQLContext = SparkSQLEnv.sqlCont
       }
       context.sparkContext.setJobDescription(substitutorCommand)
       val execution = context.sessionState.executePlan(context.sql(command).logicalPlan)
-      hiveResponse = SQLExecution.withNewExecutionId(execution, Some("cli")) {
-        hiveResultString(execution.executedPlan)
+      execution.logical match {
+        case CommandResult(_, _, _, _) =>
```
Comment (cloud-fan) [inline sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/SparkSQLDriver.scala:70]: can we add some comments to explain it?

Comment (cloud-fan) [inline sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/SparkSQLDriver.scala:70]:
```suggestion
        case _: CommandResult =>
```

Comment (AngersZhuuuu) [reply sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/SparkSQLDriver.scala]: DOne

Comment (AngersZhuuuu) [reply sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/SparkSQLDriver.scala]: Done

**Comment by cloud-fan** [comment]
https://github.com/apache/spark/pull/40437 might be related. We want to remove `hiveResultString` from CLI and only use it in hive compatibility tests.

**Comment by AngersZhuuuu** [comment]
ping @cloud-fan

**Review by jaceklaskowski — APPROVED** [review]

**Review by dongjoon-hyun — COMMENTED** [review]
cc @sunchao because he is also an expert in this domain.

File: sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/SparkSQLDriver.scala [hunk 2]
```diff
@@ -65,8 +66,15 @@ private[hive] class SparkSQLDriver(val context: SQLContext = SparkSQLEnv.sqlCont
       }
       context.sparkContext.setJobDescription(substitutorCommand)
       val execution = context.sessionState.executePlan(context.sql(command).logicalPlan)
-      hiveResponse = SQLExecution.withNewExecutionId(execution, Some("cli")) {
-        hiveResultString(execution.executedPlan)
+      // Command type sql have been executed when call `context.sql(command).logicalPlan`,
```
Comment (sunchao) [inline sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/SparkSQLDriver.scala:69]: this comment is not very easy to understand: what is `Command type sql"? maybe:
```
The SQL command has been executed above via `executePlan`, therefore we don't need to wrap it again
with a new execution ID when getting Hive result
```
?

**Review by sunchao — APPROVED** [review]
Looks OK to me. It'd be better if we can update the PR description to explain more why this PR is needed and what changes are proposed here.

File: sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/SparkSQLDriver.scala (hunk 2 above)
Comment (AngersZhuuuu) [reply sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/SparkSQLDriver.scala]: Done

**Comment by sunchao** [comment]
Merged to master, thanks!

**Pull request closed without merging**